BridgeSDK/BridgeAPI/BridgeAPITypes/SBBScheduledActivity.m

```diff
@@ -115,4 +115,39 @@ - (void)reconcileWithDictionaryRepresentation:(NSDictionary *)dictionary objectM
     }
 }
 
+#pragma mark - Used in predicates
+
+- (BOOL) isCompleted
+{
+    return self.finishedOn != nil;
+}
+
+- (BOOL) isExpired
+{
+    return (self.expiresOn != nil) && ([[NSDate date] earlierDate:self.expiresOn] == self.expiresOn);
+}
+
+- (BOOL) isAvailableNow
+{
+    return (self.scheduledOn.timeIntervalSinceNow < 0) && !self.isExpired;
+}
+
+- (NSString *) activityIdentifier
+{
+    return self.activity.task.identifier ?: self.activity.survey.identifier ?: self.activity.compoundActivity.identifier;
+}
+
+- (NSString *) scheduleIdentifier
+{
+    NSRange range = [self.guid rangeOfString:@":"];
+    if (range.location != NSNotFound)
+    {
+        return [self.guid substringToIndex:range.location];
```

Comment:
This is actually the same as self.activity.guid, I'm like 97% sure...

There's also a self.schedulePlanGuid member now, that has the guid of the schedule plan that generated this scheduled activity, which might be what you're actually looking for though?

Reply:
No, the `schedulePlanGuid` is for a different object. 

There is a one-to-many for the schedulePlan to scheduledActivity, and a one-to-one for the scheduledActivity to Activity. Might be that the self.activity.guid is the same thing, but... if it ain't broke... ;)